ast/src/analyzed/mod.rs

```diff
@@ -1267,8 +1267,10 @@ pub struct PolynomialReference {
     /// The type arguments if the symbol is generic.
     /// Guaranteed to be Some(_) after type checking is completed.
     pub type_args: Option<Vec<Type>>,
-    ///
+    /// Keep track of valid implementations for this reference, indexed by their type arguments.
     pub resolved_impls: BTreeMap<Vec<Type>, usize>,
+    /// If resolved implementation is matched with this reference, this is the name of the implemented trait.
```

Comment:
Maybe we should combine this with the one above in a struct? It is only relevant for trait function references and we could abstract that away.